powerstore/resource_host.go

```diff
@@ -578,6 +579,24 @@ func (r *resourceHost) ValidateConfig(ctx context.Context, req resource.Validate
 				"`chap_mutual_username` cannot pe present without `chap_single_username`",
 			)
 		}
+		if initiator.ChapMutualUsername == types.StringNull() && initiator.ChapMutualPassword != types.StringNull() {
+			resp.Diagnostics.AddError(
+				"Error validating config host",
+				"`chap_mutual_password` cannot pe present without `chap_mutual_username`",
+			)
+		}
+		if initiator.ChapSingleUsername == types.StringNull() && initiator.ChapSinglePassword != types.StringNull() {
+			resp.Diagnostics.AddError(
+				"Error validating config host",
+				"`chap_single_password` cannot pe present without `chap_single_username`",
+			)
+		}
+		if r.getPortType(initiator.PortName.ValueString()) != string(gopowerstore.InitiatorProtocolTypeEnumISCSI) && !initiator.ChapSingleUsername.IsNull() {
```

Comment:
can we add this validation for all chap parameters. if we specify chap_mutual_username it will ask for chap_single_username and then will throw this error.